File: sdks/ts/src/sdk/models/operations/putmetadatatemplatesenterprisesecurityclassification6vmvochwuwoschemanumberupdate.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



// PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyDataClassification
/** 
 * Additional details for the classification.
**/
export class PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyDataClassification extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=classificationDefinition" })
  classificationDefinition?: string;

  @SpeakeasyMetadata({ data: "json, name=colorID" })
  colorID?: number;
}


// PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyData
/** 
 * The details of the updated classification.
**/
export class PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyData extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=classification" })
  classification?: PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyDataClassification;

  @SpeakeasyMetadata({ data: "json, name=key" })
  key: string;
}


// PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBody
/** 
 * A single classification to update.
**/
export class PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data" })
  data: PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBodyData;

  @SpeakeasyMetadata({ data: "json, name=enumOptionKey" })
  enumOptionKey: string;

  @SpeakeasyMetadata({ data: "json, name=fieldKey" })
  fieldKey: string;

  @SpeakeasyMetadata({ data: "json, name=op" })
  op: string;
}


export class PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json-patch+json", elemType: PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBody })
  request?: PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateRequestBody[];
}


export class PutMetadataTemplatesEnterpriseSecurityClassification6VmVochwUWoSchemaNumberUpdateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  classificationTemplate?: shared.ClassificationTemplate;

  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/putmetadatatemplatesenterprisesecurityclassification6vmvochwuwoschemanumberupdate.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

// PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyDataClassification
// Additional details for the classification.
type PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyDataClassification struct {
	ClassificationDefinition *string  `json:"classificationDefinition,omitempty"`
	ColorID                  *float64 `json:"colorID,omitempty"`
}

// PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyData
// The details of the updated classification.
type PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyData struct {
	Classification *PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyDataClassification `json:"classification,omitempty"`
	Key            string                                                                                                          `json:"key"`
}

// PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBody
// A single classification to update.
type PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBody struct {
	Data          PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBodyData `json:"data"`
	EnumOptionKey string                                                                                           `json:"enumOptionKey"`
	FieldKey      string                                                                                           `json:"fieldKey"`
	Op            string                                                                                           `json:"op"`
}

type PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequest struct {
	Retries *utils.RetryConfig
	Request []PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateRequestBody `request:"mediaType=application/json-patch+json"`
}

type PutMetadataTemplatesEnterpriseSecurityClassification6VMVochwUWoSchemaNumberUpdateResponse struct {
	ClassificationTemplate *shared.ClassificationTemplate
	ClientError            *shared.ClientError
	ContentType            string
	StatusCode             int64
}
